x-pack/solutions/search/plugins/search_playground/server/lib/fetch_query_source_fields.ts

```diff
@@ -170,6 +171,40 @@ const sortFields = (fields: FieldCapsResponse['fields']): FieldCapsResponse['fie
   return Object.fromEntries(entries);
 };
 
+const createSemanticTextFieldsList = (
+  mappingProperties: Record<string, MappingProperty>,
+  semanticFieldsList: SemanticField[] = [],
+  semanticTextField?: string
+): SemanticField[] => {
+  Object.keys(mappingProperties || {}).forEach((field) => {
+    if (mappingProperties[field].type === 'semantic_text') {
+      const mapping = mappingProperties[field] as unknown as MappingSemanticTextProperty;
+      const semanticField: SemanticField = {
+        field: semanticTextField ? `${semanticTextField}.${field}` : field,
+        inferenceId: mapping?.inference_id,
+        embeddingType: mapping?.model_settings?.task_type
+          ? EMBEDDING_TYPE[mapping.model_settings.task_type]
+          : undefined,
+      };
+      semanticFieldsList.push(semanticField);
+      return semanticFieldsList;
+    } else {
+      let fieldsOrProperties: Record<string, MappingProperty> = {};
+
+      if ('fields' in mappingProperties[field]) {
+        fieldsOrProperties = (mappingProperties[field] as unknown as any).fields;
```

Comment:
we should not be casting in runtime code like this. that should only be done in tests when necessary.

Instead we should write some guard utility functions to verify the types instead of casting.

Reply:
Added this to fix check type failure.

Reply:
will try adding guard utility functions.

Reply:
updated in e254ee0cf75d5f582acb5dd35ae93c559c0b4e8f